internal/pkg/skuba/kubernetes/node_version.go

```diff
@@ -179,29 +180,47 @@ func nodeVersioningInfo(client clientset.Interface, nodeName string) (NodeVersio
 
 	// find out the container image tags, depending on the role of the node
 	if IsControlPlane(nodeObject) {
-		apiServerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-apiserver", nodeName))
-		if err != nil {
-			return NodeVersionInfo{}, errors.Wrap(err, "could not retrieve apiserver pod")
-		}
-		controllerManagerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-controller-manager", nodeName))
-		if err != nil {
-			return NodeVersionInfo{}, errors.Wrap(err, "could not retrieve controller-manager pod")
-		}
-		schedulerTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "kube-scheduler", nodeName))
-		if err != nil {
-			return NodeVersionInfo{}, errors.Wrap(err, "could not retrieve scheduler pod")
-		}
-		etcdTag, err := getPodContainerImageTag(client, metav1.NamespaceSystem, fmt.Sprintf("%s-%s", "etcd", nodeName))
+		err := skubautil.Retry(5, time.Second, func() error {
```

Comment:
Retrying the whole block will make all steps inside the block being retried. I think we should only retry on a per-call basis, so what we already retrieved is not retried if a later step fails.

Reply:
but...we need to retry the api call top get a fresh status and that means that some of the pods can have a state change meanwhile. We cant guaranteee that after X retries the pods that we checked and __were__ ok are __now__ ok. 

Either all of them are ok at the time of the check, or we need to recheck them. Performance wise, shouldnt be that big of a problem as we already have the list, so Im not sure why we would avoid rechecking them all.

Plus...this means we can move later to check their actual status (Running for example) and in that case we __really__ need all of them to be on that state at the same time, not later.

I can avoid checking already checked ones, but I dont think its such a good idea, could lead to issues down the line where we check pod X and it was ok but after X retries due to pod Y, it was no longer ok and we would no think this was the issue.